### ISSUE TITLE
[Classic] Gnomish Mind Control Cap 

### ISSUE THREAD

**Issue by Kelzior** [cmangos/issues#639]
Core 2644

Gnomish Mind Control Cap ( http://db.vanillagaming.org/?spell=12907 ) - target is not cotrolled.
The effect should be not priest's MC ( as many pirate server implement ), but target has to become "guardian" of spell casters.
Watch https://www.youtube.com/watch?v=Mh_cLN66zHA#t=503 for confirmation

**Comment by Neotmiren** [comment]
I have fixed Gnomish Mind Control Cap along with Universal Remote, Gnomish Death Ray, Gnomish Shrink Ray, Gnomish Battle Chicken just a few days ago. The Cap requires very simple code to be added to Spell::EffectDummy()

```cpp
case 13180:   // Gnomish Mind Control Cap - в бою применять нельзя; нужен флаг SPELL_ATTR_EX3_NO_INITIAL_AGGRO в attributesEx3 для spell 13180
{
    if (!unitTarget)
        return;

    int32 failureChance = 5; // TBC: unitTarget->getLevel() > 60 ? 20 : 5;  // guessed chance of failure
    if (urand(0, 99) > failureChance)
        m_caster->CastSpell(unitTarget, 13181, true);
    else
        unitTarget->CastSpell(m_caster, 13181, true);

    return;
}
````

but this code (borrowed from TBC HG-core) is insufficient because of the same problem as with Mind Soothe and Soothe Animal spells. These spells do not have SPELL_ATTR_EX3_NO_INITIAL_AGGRO flag set in spell.dbc data

Maybe the equivalent of SPELL_ATTR_EX3_NO_INITIAL_AGGRO flag could be guessed based on other parameters of a spell, but for now I fix such things by modifying spell.dbc (Classic version, 16305013 bytes)

00697C72: 00 02 ; id 13180

If someone is interested in other mentioned items, as well as in much more significant improvements, like proper chase movement of NPC spellcasters/ranged attackers (eliminates easy LoS exploit, when they just stand behind the wall and wait while you defeat melee NPCs), or making water-only creatures not to leave combat when player rises to surface (even Levitates) etc... try to borrow source code from downloads available from URL in my profile. I made all mentioned changes just recently and probably will upload .7z archives of binary repacks (including sources of CMangos modifications) soon, after some practice.
